### DISCUSSION TITLE
Auzure client credentails configuration issue

### DISCUSSION THREAD

**Discussion by vamsinm** [openfga/community#323] [General]
Openfga is running with OIDC authentication method using Azure client id and client secret. 

Environment variables :
OPENFGA_AUTHN_METHOD = oidc
OPENFGA_DATASTORE_ENGINE = postgres
OPENFGA_DATASTORE_URI = 'postgres://<postgressurl>/postgres'
OPENFGA_DATASTORE_USERNAME = postgres
OPENFGA_OIDC_ISSUER_URL = https://login.microsoftonline.com/<azureid>/v2.0
OPENFGA_AUTHN_OIDC_ISSUER = https://login.microsoftonline.com/<azureid>/v2.0
OPENFGA_AUTHN_OIDC_AUDIENCE = oidcaudience
 OPENFGA_OIDC_CLIENT_ID= oidc client id
OPENFGA_OIDC_CLIENT_SECRET = client secret
OPENFGA_DATASTORE_PASSWORD = postgress password

trying to connect with clientcrednetails using the example: https://openfga.dev/docs/getting-started/setup-sdk-client#using-client-credentials-flow and seeing error : failed to get list of stores with error ListStores auth error for GET ListStores with body {"code":"auth_failed_invalid_bearer_token","message":"invalid bearer token"}

am i missing anything? thank you

openfga version 1.5.3
fga version v`0.4.0`

**Comment by ewanharris** [comment]
👋🏻 Hey @vamsinm,

It looks like the error is arising from when [OpenFGA is parsing the JWT](https://github.com/openfga/openfga/blob/d0d74b90fe7dd1ea1e4c0afe48f527f75ca711e8/internal/authn/oidc/oidc.go#L74-L81) provided as the auth header. Unfortunately there's not much specific as to what failed there and we don't log the specific error.

I'd suggest adding a log of the [error here](https://github.com/openfga/openfga/blob/d0d74b90fe7dd1ea1e4c0afe48f527f75ca711e8/internal/authn/oidc/oidc.go#L79) and building and running OpenFGA locally to provide better insight into the error you're getting with that token.

**Reply by vamsinm** [reply]
wondering if anyone has working  example of openfga with azure and client call with azure. thank you!

**Reply by vamsinm** [reply]
thank you  for suggestion. able to print error that helped to resolve.
